File: react-native/storyly-placement-react-native/android/src/main/java/com/storylyplacementreactnative/common/RNPlacementEventType.kt
package com.storylyplacementreactnative.common


enum class RNPlacementEventType(val eventName: String) {
    ON_WIDGET_READY("onWidgetReady"),
    ON_FAIL("onFail"),
    ON_EVENT("onEvent"),
    ON_ACTION_CLICKED("onActionClicked"),
    ON_PRODUCT_EVENT("onProductEvent"),
    ON_CART_UPDATE("onCartUpdate"),
    ON_WISHLIST_UPDATE("onWishlistUpdate"),
}

File: react-native/storyly-placement-react-native/ios/Common/RNPlacementEventType.swift
import Foundation

@objc public enum RNPlacementEventType: Int {
  case onWidgetReady
  case onFail
  case onEvent
  case onActionClicked
  case onProductEvent
  case onCartUpdate
  case onWishlistUpdate
  
  public var eventName: String {
    switch self {
    case .onWidgetReady:
      return "onWidgetReady"
    case .onFail:
      return "onFail"
    case .onEvent:
      return "onEvent"
    case .onActionClicked:
      return "onActionClicked"
    case .onProductEvent:
      return "onProductEvent"
    case .onCartUpdate:
      return "onCartUpdate"
    case .onWishlistUpdate:
      return "onWishlistUpdate"
    }
  }
}

